web-api/src/main/java/com/graphhopper/ResponsePath.java

```diff
@@ -119,11 +120,22 @@ public PointList getWaypoints() {
     /**
      * This method initializes this path with the snapped input points.
      */
-    public void setWaypoints(PointList wpList) {
+    public ResponsePath setWaypoints(PointList wpList) {
         if (waypointList != PointList.EMPTY)
             throw new IllegalStateException("Cannot call setWaypoints twice");
 
         this.waypointList = wpList;
+        return this;
+    }
+
+    public List<Interval> getWaypointIntervals() {
+        check("getWaypointIntervals");
+        return waypointIntervals;
+    }
```

Comment:
It could be useful to add these intervals to the server response, but this needs to be done manually in `ResponsePathSerializer`.

Reply:
Will do in another PR.